src/main/java/core/basesyntax/ReadWriteList.java

```diff
@@ -2,24 +2,40 @@
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.concurrent.locks.Lock;
 import java.util.concurrent.locks.ReadWriteLock;
 import java.util.concurrent.locks.ReentrantReadWriteLock;
 
 public class ReadWriteList<E> {
     private List<E> list = new ArrayList<>();
-    private ReadWriteLock lock = new ReentrantReadWriteLock();
+    private final ReadWriteLock lock = new ReentrantReadWriteLock();
+    private final Lock writeLock = lock.writeLock();
+    private final Lock readLock = lock.readLock();
 
     public void add(E element) {
```

Comment:
Only one thread can update the data at a time causing other threads (both readers and writers) to block until the update is over.

Reply:
This means that you have to block reading while writing as well

Reply:
understood))